Sources/Basic/Thread.swift

```diff
@@ -32,12 +32,16 @@ final public class Thread {
         finishedCondition = Condition()
 
         // Wrap the task with condition notifying any other threads blocked due to this thread.
-        let theTask = { [unowned self] in
-            self.finishedCondition.lock()
-            defer { self.finishedCondition.unlock() }
+        // Capture self weakly to avoid reference cycle. In case Thread is deinited before the task
+        // runs, skip the use of finishedCondition.
+        let theTask = { [weak self] in
+            self?.finishedCondition.lock()
+            defer { self?.finishedCondition.unlock() }
             task()
-            self.finished = true
-            self.finishedCondition.broadcast()
+            if let strongSelf = self {
+                strongSelf.finished = true
+                strongSelf.finishedCondition.broadcast()
+            }
```

Comment:
I would be tempted to write:
```
if let strongSelf = self {
  precondition(!strongSelf.finishedCondition)
  strongSelf.finishedCondition.whileLocked {
    task()
    strongSelf.finished = true
    strongSelf.finishedCondition.broadcast()
  }
} else {
  // If the containing thread has been destroyed, we can ignore the finished condition and just run the task.
  task()
}  
```
so it is more obvious exactly operations can be performed as a group...